module/modmanager/manager.go

```diff
@@ -123,9 +168,10 @@ func (mgr *Manager) Close(ctx context.Context) error {
 		mgr.restartCtxCancel()
 	}
 	var err error
-	for _, mod := range mgr.modules {
+	mgr.modules.Range(func(_ string, mod *module) bool {
 		err = multierr.Combine(err, mgr.closeModule(mod, false))
-	}
+		return true
```

Comment:
the API is a little awkward - my understanding is that returning `true` is essentially like `continue` - it will stop once we go over all of the keys. returning `false` just lets us stop earlier than that.

Reply:
Got it; so if we never return `false`, we'll just look at every key and value in the map?

Reply:
yep! just did a little script to confirm:

```go
func main() {
	var m sync.Map
	m.Store("a", 1)
	m.Store("b", 2)
	m.Range(func(k any, v any) bool {
		fmt.Println(k, v)
		return true
	})
}

// output (in different orders)
// a 1
// b 2
```